jwplatform/client.py

```diff
@@ -219,28 +256,81 @@ def clip(self, site_id, channel_id, event_id, body=None, query_params=None):
         )
 
 
-class _MediaClient(_SiteResourceClient):
+CREATE_MEDIA_PAYLOAD = {
+    "upload": {
+    },
+    "metadata": {
+
+    }
+}
 
+
+class _MediaClient(_SiteResourceClient):
     _resource_name = "media"
     _id_name = "media_id"
 
+    def __init__(self, client: JWPlatformClient):
+        super().__init__(client)
+
     def reupload(self, site_id, body, query_params=None, **kwargs):
         resource_id = kwargs[self._id_name]
         return self._client.request(
             method="PUT",
-            path=self._singular_path.format(site_id=site_id, resource_name=self._resource_name, resource_id=resource_id) + "reupload/",
+            path=self._singular_path.format(site_id=site_id, resource_name=self._resource_name,
+                                            resource_id=resource_id) + "reupload/",
             body=body,
             query_params=query_params
         )
 
-class _WebhookClient(_ResourceClient):
+    def get_upload_handler(self, site_id, file, body=None, query_params=None, **kwargs):
+        if not kwargs:
+            kwargs = {}
+
+        # Determine the upload type - Single or multi-part
+        target_part_size = int(kwargs['target_part_size']) if 'target_part_size' in kwargs else constants.MIN_PART_SIZE
+        upload_method = determine_upload_method(file, target_part_size)
+        if not body:
+            body = CREATE_MEDIA_PAYLOAD.copy()
+        body["upload"]["method"] = upload_method
```

Comment:
What should happen if the body was specified but didn't have an `"upload"` key, or `"upload"` wasn't a `dict`?